elrond-wasm-node/src/api/send_api_node_managed.rs

```diff
@@ -0,0 +1,500 @@
+        }
+    }
+
+    fn deploy_contract(
+        &self,
+        gas: u64,
+        amount: &BigUint<Self::ProxyTypeManager>,
+        code: &ManagedBuffer<Self::ProxyTypeManager>,
+        code_metadata: CodeMetadata,
+        arg_buffer: &ManagedArgBuffer<Self::ProxyTypeManager>,
+    ) -> Option<ManagedAddress<Self::ProxyTypeManager>> {
+        unsafe {
+            let code_metadata_handle = code_metadata_to_buffer_handle(code_metadata);
+            let new_address_handle = mBufferNew();
+            let result_handle = mBufferNew();
+            let _ = managedCreateContract(
+                gas as i64,
+                amount.get_raw_handle(),
+                code.get_raw_handle(),
+                code_metadata_handle,
+                arg_buffer.get_raw_handle(),
+                new_address_handle,
+                result_handle,
+            );
+
+            Some(ManagedAddress::from_raw_handle(
+                self.clone(),
+                new_address_handle,
+            ))
+        }
+    }
+
+    fn deploy_from_source_contract(
+        &self,
+        gas: u64,
+        amount: &BigUint<Self::ProxyTypeManager>,
+        source_contract_address: &ManagedAddress<Self::ProxyTypeManager>,
+        code_metadata: CodeMetadata,
+        arg_buffer: &ManagedArgBuffer<Self::ProxyTypeManager>,
+    ) -> Option<ManagedAddress<Self::ProxyTypeManager>> {
+        unsafe {
+            let code_metadata_handle = code_metadata_to_buffer_handle(code_metadata);
+            let new_address_handle = mBufferNew();
+            let result_handle = mBufferNew();
+            let _ = managedDeployFromSourceContract(
+                gas as i64,
+                amount.get_raw_handle(),
+                source_contract_address.get_raw_handle(),
+                code_metadata_handle,
+                arg_buffer.get_raw_handle(),
+                new_address_handle,
+                result_handle,
+            );
+            // TODO: return result
+
+            Some(ManagedAddress::from_raw_handle(
+                self.clone(),
+                new_address_handle,
+            ))
+        }
+    }
+
+    fn upgrade_contract(
+        &self,
+        sc_address: &ManagedAddress<Self::ProxyTypeManager>,
+        gas: u64,
+        amount: &BigUint<Self::ProxyTypeManager>,
+        code: &ManagedBuffer<Self::ProxyTypeManager>,
+        code_metadata: CodeMetadata,
+        arg_buffer: &ManagedArgBuffer<Self::ProxyTypeManager>,
+    ) {
+        unsafe {
+            let code_metadata_handle = code_metadata_to_buffer_handle(code_metadata);
+            let result_handle = mBufferNew();
+            managedUpgradeContract(
+                sc_address.get_raw_handle(),
+                gas as i64,
+                amount.get_raw_handle(),
+                code.get_raw_handle(),
+                code_metadata_handle,
+                arg_buffer.get_raw_handle(),
+                result_handle,
+            );
+
+            // TODO: return result
```

Comment:
TODO reminder, in case you wanted to do it later in this PR.

Reply:
Next PR, I also need to ad tests